elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -90,64 +97,81 @@ public ActivePermissionExecutor(final com.yahoo.elide.core.RequestScope requestS
      * @see com.yahoo.elide.annotation.DeletePermission
      */
     @Override
-    public <A extends Annotation> void checkPermission(Class<A> annotationClass,
+    public <A extends Annotation> ExpressionResult checkPermission(Class<A> annotationClass,
                                                        PersistentResource resource,
                                                        ChangeSpec changeSpec) {
         if (requestScope.getSecurityMode() == SecurityMode.SECURITY_INACTIVE) {
-            return; // Bypass
+            return ExpressionResult.PASS; // Bypass
         }
+
         Expressions expressions;
         if (SharePermission.class == annotationClass) {
             expressions = expressionBuilder.buildSharePermissionExpressions(resource);
```

Comment:
SharePermission seems to be a special case. If I bypass sharepermission check, [this shareable test](https://github.com/yahoo/elide/blob/master/elide-integration-tests/src/test/groovy/com/yahoo/elide/tests/ShareableIT.groovy#L63-L76) will fail